src/mudlet-lua/lua/Other.lua

```diff
@@ -700,3 +700,111 @@ function condenseMapLoad()
 
   return loadtime
 end
+
+do
+  -- management things
+
+  -- Dictionary with events as keys and lists of lua functions as values to dispatch events
+  -- to the right functions.
+  local handlers = {}
+
+  -- Remember highest hander ID to avoid ID reuse.
+  local highestHandlerId = 0
+  -- Helps us finding the right event handler from an ID.
+  local handlerIdsToHandlers = {}
+
+  -- C functions that get overwritten.
+  local origRegisterAnonymousEventHandler = registerAnonymousEventHandler
+
+  function registerAnonymousEventHandler(event, func, isOneShot)
+    if type(event) ~= "string"then
+      error(
+        string.format(
+          "Unexpected argument type in function registerAnonymousEventHandler for argument #1. String expected, got %s.",
```

Comment:
Do we want to use the _same_ message format as the _internal_ (C++-core implemented one) would do, i.e.:

    "registerAnonymousEventHandler: bad argument #1 type (event name as string expected, got %s!)"

Reply:
Probably. I didn't really look the actual message up (I wanted to do that later, I remember now).

Reply:
I agree, lets align it with:

```                            "raiseEvent: bad argument #%d type (string, number, boolean, or nil expected, got a %s!)"```